pkpdapp/pkpdapp/forms/dataset.py

```diff
@@ -15,58 +20,219 @@ def file_size(value):
+                    description="",
+                    unit=unit,
+                    dataset=instance)
 
-class CreateNewBiomarkerUnit(forms.Form):
+        biomarker_index = {}
+        for i, b in enumerate(bts_unique):
+            biomarker_index[b] = i
+        # save each row of data as either biomarker or dose
+        for index, row in data.iterrows():
+            if row['TIME_UNIT'] != ".":
+                time_unit = Unit.objects.get(symbol=row['TIME_UNIT'])
+            else:  # assume hours as default
+                time_unit = Unit.objects.get(symbol='h')
+            value = row['DV']
+            subject_id = row['ID']
+            if value != ".":  # measurement observation
+                Biomarker.objects.create(
+                    time=time_unit.multiplier * row['TIME'],
+                    subject_id=subject_id,
+                    value=row['DV'],
+                    biomarker_type=BiomarkerType.objects.get(
+                        name=row['YDESC'],
+                        dataset=instance)
+                )
+            else:  # dose observation
+                compound_str = row['COMPOUND']
+                try:
+                    compound = Compound.objects.get(name=compound_str)
+                except Compound.DoesNotExist:
+                    compound = Compound.objects.create(
+                        name=compound_str
+                    )
+                try:
+                    protocol = Protocol.objects.get(
+                        dataset=instance,
+                        subject_id=subject_id,
+                        compound=compound
+                    )
+                except Protocol.DoesNotExist:
+                    protocol = Protocol.objects.create(
+                        name='{}-{}-{}'.format(
+                            instance.name,
+                            compound.name,
+                            subject_id
+                        ),
+                        compound=compound,
+                        dataset=instance,
+                        subject_id=subject_id,
+                    )
+                start_time = time_unit.multiplier * float(row['TIME'])
+                amount = float(row['AMT'])
+                Dose.objects.create(
+                    start_time=start_time,
+                    amount=amount,
+                    protocol=protocol,
+                )
+
+        return instance
+
+
+class CreateNewBiomarkerType(forms.ModelForm):
     """
     A form to associate a unit with a predefined biomarker type name.
     """
-    UNIT_CHOICES = [
-        ('mg', 'mg'),
-        ('g', 'g'),
-        ('cm3', 'cm^3'),
-    ]
-    unit = forms.ChoiceField(label='', choices=UNIT_CHOICES)
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    class Meta:
+        model = BiomarkerType
+        fields = ['description']
+
+    symbol = forms.CharField(label='Unit', required=False)
```

Comment:
Probably want a `django.ModelChoiceField` here for `Unit`

Reply:
Done.